sort_images.sh

```diff
@@ -0,0 +1,165 @@
+#! /usr/bin/env bash
+
+#----------------------------------------------------------------------
+# NAME: sort_images.sh
+#
+# BRIEF: Command-line Bash script with prompts to help sort photo files.
+# - creates & sorts image files into year & month subdirectories
+# - also creates day subdirectories, if answer at prompt is `Y` for Yes
+# - sorts by photo-taken date or if missing, by modify date - since sometimes
+# creation date is assigned as file download date, which is incorrect.
+#
+# AUTHOR: Kim Lew
+#----------------------------------------------------------------------
+
+if [ $# -eq 0 ]; then
+  echo "Type the directory path that needs Year & Month subdirectories."
+  echo "Leave off trailing / at end of path: "
+  read -r directory_path
+  echo "Do you also want a Day subdirectory? (Y for Yes, N for No):"
+  read -r day_subdir_also
+else
+  if [ ! "$1" ]; then
+    echo "Enter the directory path - as the 1st parameter."
+    echo "Enter Y for Yes or N for No - for Day subdirectory as 2nd parameter."
+    exit 1
+  fi
+fi
+
+if [[ $# -eq 1 || $# -eq 2 ]]; then
+  directory_path="$1"
+  safe_day_subdir_also='n'
+
+  if [ $# -eq 2 ]; then
+    day_subdir_also="$2"
+
+    case "${day_subdir_also}" in
+      [yY] | [yY][eE][sS]) # To create year-month-day subdirectories.
+        safe_day_subdir_also='y'
+    ;;
+      [nN] | [nN][oO] | '') # To create ONLY year-month subdirectories.
+        safe_day_subdir_also='n'
+    ;;
+       *)
+    echo "Invalid input. Enter Y or N. There is a problem with the 2nd parameter \
+    given or the answer to the 2nd prompt."
+    exit 1
+    ;;
+    esac
+  fi
+elif [ $# -gt 2 ]; then 
+  # Case of 3 or more parameters given.
+  echo "Give 1 or 2 command-line arguments. Or give 0 arguments & get prompts."
+  exit 1
+else # No command-line arguments given. Uses prompts & with checks.
+  continue
+fi
```

Comment:
This clause results in an error message from `bash` when there are no command-line arguments:

```
../sort_images.sh: line 55: continue: only meaningful in a `for', `while', or `until' loop
```

The `continue` statement has to be in the body of a loop. Here, you can just remove the `else` clause and end with a `fi`, since you aren't going to do anything in this case.

Reply:
@kcharter Removed.

Reply:
OK, the change looks good and I have tried it out. Works like a charm!